tests/conftest.py

```diff
@@ -119,3 +119,75 @@ def mock_mappings(tmp_path):
         'bar_centric': {'mapping': bar_mapping, 'settings': bar_settings}
     }
     return root, expected
+
+
+def mock_listdir_nodefs(models_dir):
+    if os.path.basename(models_dir) == 'es-models': 
+        return ['gdc_from_graph']
+    elif os.path.basename(models_dir) == 'gdc_from_graph':
+        return ['case.mapping.yaml']
+
+
+def mock_load_yaml_for_defs(filename):
+    name = os.path.basename(filename)
+    if name == 'case.mapping.yaml':
+        return {'properties': 'someproperty'}
+    elif name == 'definitions.yaml':
+        return {'_meta': 'expected_result'} 
+
+def mock_load_yaml_missing_definitions(filename):
+    name = os.path.basename(filename)
+    if name == 'case.mapping.yaml':
+        return {'properties': 'someproperty'}
+    elif name == 'definitions.yaml':
+        return yaml.load(open('this_file_doesnt_exist', 'r'))
+
+
+def mock_load_yaml_empty_definitions(filename):
+    name = os.path.basename(filename)
+    if name == 'case.mapping.yaml':
+        return {'properties': 'someproperty'}
+    elif name == 'definitions.yaml':
+        return {}
+
+
+def mock_isfile_no_def(filename):
+    if os.path.basename(filename) == 'gdc_from_graph':
+        return False
+    elif os.path.basename(filename) == 'definitions.yaml':
+        return False
+    return True
+
+
+def mock_isfile_def(filename):
+    if os.path.basename(filename) == 'gdc_from_graph':
+        return False
+    elif os.path.basename(filename) == 'definitions.yaml':
+        return True
+    return True
+
+
+@pytest.fixture
+def add_def_file(monkeypatch):
```

Comment:
can we mock the filesystem instead of mocking calls? similar to what's being done in `mock_mappings` fixture?

Reply:
you got it boss